rust/datafusion/src/execution/physical_plan/hash_aggregate.rs

```diff
@@ -677,6 +532,152 @@ impl RecordBatchReader for HashAggregateIterator {
+                ))
+            }
+        };
+        builders.push(builder);
+    }
+
+    // iterate over the map
+    for (k, v) in map.iter() {
+        // add group values to builders
+        for i in 0..num_group_expr {
+            let builder = builders[i].as_any_mut();
+            match &k[i] {
+                GroupByScalar::Int8(n) => append_group_value!(builder, Int8Builder, *n),
+                GroupByScalar::Int16(n) => append_group_value!(builder, Int16Builder, *n),
+                GroupByScalar::Int32(n) => append_group_value!(builder, Int32Builder, *n),
+                GroupByScalar::Int64(n) => append_group_value!(builder, Int64Builder, *n),
+                GroupByScalar::UInt8(n) => append_group_value!(builder, UInt8Builder, *n),
+                GroupByScalar::UInt16(n) => {
+                    append_group_value!(builder, UInt16Builder, *n)
+                }
+                GroupByScalar::UInt32(n) => {
+                    append_group_value!(builder, UInt32Builder, *n)
+                }
+                GroupByScalar::UInt64(n) => {
+                    append_group_value!(builder, UInt64Builder, *n)
+                }
+                GroupByScalar::Utf8(str) => {
+                    append_group_value!(builder, StringBuilder, str)
+                }
+            }
+        }
+
+        // add agggregate values to builders
+        for i in 0..num_aggr_expr {
+            let value = v[i].borrow().get_value()?;
+            let index = num_group_expr + i;
+            let builder = builders[index].as_any_mut();
+            match output_schema.field(i).data_type() {
+                DataType::Int8 => append_aggr_value!(builder, Int8Builder, value, Int8),
+                DataType::Int16 => {
+                    append_aggr_value!(builder, Int16Builder, value, Int16)
+                }
+                DataType::Int32 => {
+                    append_aggr_value!(builder, Int32Builder, value, Int32)
+                }
+                DataType::Int64 => {
+                    append_aggr_value!(builder, Int64Builder, value, Int64)
+                }
+                DataType::UInt8 => {
+                    append_aggr_value!(builder, UInt8Builder, value, UInt8)
+                }
+                DataType::UInt16 => {
+                    append_aggr_value!(builder, UInt16Builder, value, UInt16)
+                }
+                DataType::UInt32 => {
+                    append_aggr_value!(builder, UInt32Builder, value, UInt32)
+                }
+                DataType::UInt64 => {
+                    append_aggr_value!(builder, UInt64Builder, value, UInt64)
+                }
+                DataType::Float32 => {
+                    append_aggr_value!(builder, Float32Builder, value, Float32)
+                }
+                DataType::Float64 => {
+                    append_aggr_value!(builder, Float64Builder, value, Float64)
+                }
+                DataType::Utf8 => {
```

Comment:
I am sure you have a good reason, but I didn't see quite why this match arm can't use `append_aggr_value!` as well

Reply:
The issue here was that `ScalarValue::Utf8` contains `String` and the builder wants `&str`. In all other cases the scalar and builder types are the same.

Reply:
that might be worth adding in a comment so future readers would not have to wonder.

Reply:
Done.